src/common/telemetry_ops/app_telemetry.rs

```diff
@@ -130,6 +134,38 @@ fn get_system_data() -> RunningEnvironmentTelemetry {
         ram_size: sys_info::mem_info().ok().map(|x| x.total as usize),
         disk_size: sys_info::disk_info().ok().map(|x| x.total as usize),
         cpu_flags: cpu_flags.join(","),
+        cpu_endian_compiled: Some(CpuEndian::compiled()),
+        cpu_endian_runtime: Some(CpuEndian::runtime()),
+    }
+}
+
+#[derive(Serialize, Clone, Copy, Debug, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum CpuEndian {
+    Little,
+    Big,
+    Other,
+}
+
+impl CpuEndian {
+    /// Compile time byte order
+    pub const fn compiled() -> Self {
+        if cfg!(target_endian = "little") {
+            CpuEndian::Little
+        } else if cfg!(target_endian = "big") {
+            CpuEndian::Big
+        } else {
+            CpuEndian::Other
+        }
+    }
+
+    /// Runtime byte order
+    pub fn runtime() -> Self {
+        match cpu_endian::working() {
+            cpu_endian::Endian::Little => CpuEndian::Little,
+            cpu_endian::Endian::Big => CpuEndian::Big,
+            cpu_endian::Endian::Minor => CpuEndian::Other,
+        }
```

Comment:
I think both of them always be the same? The crate you use perform similar compile-time check as you did, but through C++ code.
https://github.com/wbcchsyn/rust-cpu-endian/blob/69096aa63d63f065dd5e809c0462ed1c4796f4d8/c_src/native_endian.cpp#L11-L20

Reply:
IIRC the crate asserts at runtime. This is relevant on arm64 where endianness can be toggled on the fly.

But, lets dive into it anyway to confirm.

Reply:
@xzfc I switched back to just a single field for compile time endianness.